index.rst

```diff
@@ -5,77 +5,331 @@
+*  A single CSC means there is a single event for each condition.
+   If the CSC is handling multiple conditions the information ends up getting "lost" in the data queue and it is not easily digestible by LOVE and other systems.
+   Whereas, if each instance of the ``Catcher`` handles a single condition/rule, events are well separated for each condition.
 
-Below we describe how some of the Catcher functionality would be implemented.
+*  In the single CSC scenario, adding/removing/updating a particular rule requires restarting the entire system.
+   In the indexed component scenario, it only requires bringing in a new instance or removing/updating an existing one.
 
-Alarms
-------
+To control the configuration of each ``Catcher`` instance, we can adopt an indexed-based configuration format, used by some CSCs in the system already.
+Basically we map an index to a CSC configuration.
+All configurations resides in the same configuration file and the CSC decides which configuration to load based on its index.
+
+Rules
+-----
+
+The first thing the ``Catcher`` must be able to do is have a set of user-defined "activation" rules.
+This basically consists of an input data stream that will be processed according to some user-defined code and be activated according to some user-define rule.
+
+For example, one might be interested in a rule that activates when a new image finishes reading (camera publishes endReadout) or something a bit more complicated like activate when a new image finishes reading, the telescope is pointing in the direction of the wind and wind speed is above a certain threshold.
+
+In terms of design, we have two different options we could implement; configuration-based or programmatic-based rules.
+
+Configuration-based rules, are based on simple rules that can be combined together to form more complex behavior.
+For example, we could a rule that receives a component name and event name and is activated every time that event is published or a slightly more complicated rule that takes 2 components/topic/attributes and check if their values are within a certain threshold.
+
+The configuration for a rule that would monitor an event would look something like this:
+
+.. code-block:: yaml
+
+   rule:
+      monitorEvent:
+         component: ATCamera
+         event: endReadout
 
-One of the key features expected from the Catcher was the ability to issue alarms that can be displayed in LOVE.
-One option to support this feature would be to create a custom LOVE producer that can be interacted with using a REST API.
-The alert information would be persisted in Sasquatch by the `Flux`_ script, which would then post a message to the LOVE producer of the new alert.
-The producer would then read the alert information from the EFD and display it.
+Combining rules would be something like:
+
+.. code-block:: yaml
+
+   rule:
+      monitorEvent:
+         component: ATCamera
+         event: endReadout
+      compareTopics:
```

Comment:
Is `compareTopics` a thing built into yaml? Or will we need to define some library of condition evaluations?

Reply:
I updated this part of the doc to expand the explanation. Hopefully it will clarify this question...

Reply:
ok.. helps a bit.. but I still don't quite get it...
don't you need an expression or sorts?
`abs(topic1-topic2)>threshold` or something like that?
OR 
is "compareTopics" the name of a "method" that contains the logic that evaluates the condition? I'm guessing you mean that! If so, the maybe change "compareTopics" to `userDefinedCondition1` or `TiagosCondition` etc... just to show that that it's defined by the user and part of the codebase.